lib/ansible/galaxy/collection.py

```diff
@@ -578,6 +578,10 @@ def verify_collections(collections, search_paths, apis, validate_certs, ignore_e
                             break
                     if local_collection is None:
                         raise AnsibleError(message='Collection %s is not installed in any of the collection paths.' % collection_name)
+                    if local_collection.latest_version == '*':
```

Comment:
It's technically possible that someone could `ansible-galaxy collection init` and then edit the galaxy.yml version to be "*", then build and publish. I feel like no one should do that, but if I should handle that gracefully then the check/error should happen after remote_collection is fetch and be something like this:
```python
if local_collection.latest_version == '*' and remote_collection.latest_version != '*':
    msg = 'Collection %s does not appear to have the MANIFEST.json. ' % collection_name
    msg += 'A MANIFEST.json is expected if the collection has been built and installed via ansible-galaxy.'
    raise AnsibleError(message=msg)
```
Other opinions welcome.

Reply:
I'm not sure they could actually publish to Galaxy. It would reject that version as not being semver.

Reply:
Okay, gtk. I only tried to build with the version "*" and was surprised it worked.